### DISCUSSION TITLE
Rust Z3. How can I convert Float to f32/f64?

### DISCUSSION THREAD

**Discussion by FoxMakarov** [Z3Prover/z3#6962] [General]
Hello, everyone!
I am learning to use z3-prover in my rust program.  But i encountered some problems.

I am using Float like in the tutorial (https://docs.rs/z3/0.10.0/z3/ast/struct.Float.html). But I noticed that it only has functions that convert from f32/f64 to Float Without converting from Float to f32/f64 function.

```
 // Create a 32-bit (IEEE-754) Float [`Ast`](trait.Ast.html) from a rust f32
pub fn from_f32(ctx: &'ctx Context, value: f32) -> Float<'ctx> 

// Create a 64-bit (IEEE-754) Float [`Ast`](trait.Ast.html) from a rust f64
pub fn from_f64(ctx: &'ctx Context, value: f64) -> Float<'ctx> 
```

I want to know if there is a method that can be used like the as_i64 function of 'Int' structure which can  convert 'Int' to the i64 variable of Rust.

How can I convert 'Float' to f32/f64?
Can you help me please?

z3 version :  0.10.0

**Comment by waywardmonkeys** [comment]
I've created https://github.com/prove-rs/z3.rs/pull/252 which would fix this ... I'll merge it into `master` over there within a day or so if I don't hear any reason not to ...

**Comment by FoxMakarov** [comment]
Thanks  for your help !!!
